### ISSUE TITLE
TBD for Meeting 21.10.20

### ISSUE THREAD

**Issue by matthaeusheer** [matthaeusheer/uncertify#8]
# Data Management
## HDF5 Dataset Processing
- [x] merge CamCAN processing possibility into BraTS processing script
- [x] possibility to exclude empty (all-zero mask) slices -> will change slices per patient so for evaluation it is nicer to have them in to loop over fixed batch_size (patient-wise) & exclude them on the fly, thus both possibilities are needed
- [x] attach all hyperparameters (args calling the processing script) into HDF5 and attach to custom dataloader for reproducibility
- [x] named hdf5 output
- [x] config class for processing for more reproducible results and greater flexibility in putting together datasets

## Datasets to create
- [x] camcan T2 & brats T2/T1 both histogram matched to same camcan sample
- [x] additionaly brats T2/T1 matched to some other sample in their distribution (s.t. they are equal amongst each other but do not match training set)
- [x] create Gaussian noise dataset
- [ ] possibility to do horizontal and vertical flip on brain datasets
- [ ] mnist upscaled -> discuss whether the upscaled makes sense at all to compare against
- [ ] could performance be improved by merging different modalities (T1, T2 together somehow?) 

# Training
- [ ] check why training on cluster is significantly slower -> profiling
- [x] check standardization vs rescaling normalization differences
- [x] train models for data defined above

# Evaluation
## Pixel-wise Anomaly Detection
- [x] NLL / KLD (/ other statistics?) histograms for datasets described above
- [x] AUROC, AUPR, DICE (at best threshold), report DICE on all slices together *and* patient-wise
## Sample-wise Anomaly Detection
- [x] Define metrics used
- [x] slice-wise and patient-wise (mean/std) reporting (check Zimmerer paper)
